app/values/resource_identifier/solr_repository.rb

```diff
@@ -0,0 +1,66 @@
+class ResourceIdentifier
+  ##
+  # Repository for querying solr for a single or multiple IDs and getting back a
+  # document object which can return its dependants and timestamps.
+  class SolrRepository
+    class << self
+      # @param [Array<Integer> OR Integer] id ID or array of IDs of records to
+      #   query for.
+      # @return [Record, #timestamp, #dependant_ids] Solr-record powered record
+      #   which can return its timestamp and dependant ids.
+      def find(id)
+        id = Array.wrap(id)
+        return CompositeRecord.new({}) if id.blank?
+        records = solr.query(solr_query(id), fl: "timestamp member_ids_ssim", sort: "timestamp desc").map do |r|
```

Comment:
To make this more efficient we would have to make filesets reindex their parents when they're deleted.